pkg/capabilities/registry/base.go

```diff
@@ -142,46 +159,241 @@ func (r *baseRegistry) Add(ctx context.Context, c capabilities.BaseCapability) e
 		return err
 	}
 
-	switch info.CapabilityType {
-	case capabilities.CapabilityTypeTrigger:
-		_, ok := c.(capabilities.TriggerCapability)
-		if !ok {
-			return errors.New("trigger capability does not satisfy TriggerCapability interface")
+	id := info.ID
+	bc, ok := r.m[id]
+	if ok {
+		switch state := bc.GetState(); state {
+		case connectivity.Shutdown, connectivity.TransientFailure:
```

Comment:
Transient is actually fatal in our case, since the plugin cannot be redialed with the same ID.